File: android/app/src/main/kotlin/com/lepos/lepos/designtokens/DesignTokens.kt
package com.lepos.lepos.designtokens

/**
 * Shared design tokens for visual consistency across platforms.
 * 
 * Android: Map to Material 3 Expressive theme
 * iOS: Map to SwiftUI Color/Spacing extensions
 */
object DesignTokens {
    
    object Colors {
        // Base Palette (Common)
        const val Primary = 0xFF6200EE
        const val PrimaryVariant = 0xFF3700B3
        const val Secondary = 0xFF03DAC6
        const val SecondaryVariant = 0xFF018786
        
        const val OnPrimary = 0xFFFFFFFF
        const val OnSecondary = 0xFF000000
    }
    
    object LightColors {
        const val Primary = 0xFF6750A4
        const val OnPrimary = 0xFFFFFFFF
        const val PrimaryContainer = 0xFFEADDFF
        const val OnPrimaryContainer = 0xFF21005D
        const val InversePrimary = 0xFFD0BCFF
        
        const val Secondary = 0xFF625B71
        const val OnSecondary = 0xFFFFFFFF
        const val SecondaryContainer = 0xFFE8DEF8
        const val OnSecondaryContainer = 0xFF1D192B
        
        const val Tertiary = 0xFF7D5260
        const val OnTertiary = 0xFFFFFFFF
        const val TertiaryContainer = 0xFFFFD8E4
        const val OnTertiaryContainer = 0xFF31111D
        
        const val Error = 0xFFB3261E
        const val OnError = 0xFFFFFFFF
        const val ErrorContainer = 0xFFF9DEDC
        const val OnErrorContainer = 0xFF410E0B
        
        const val Background = 0xFFFFFBFE
        const val OnBackground = 0xFF1C1B1F
        
        const val Surface = 0xFFFFFBFE
        const val OnSurface = 0xFF1C1B1F
        const val SurfaceVariant = 0xFFE7E0EC
        const val OnSurfaceVariant = 0xFF49454F
        const val SurfaceTint = 0xFF6750A4
        const val InverseSurface = 0xFF313033
        const val InverseOnSurface = 0xFFF4EFF4
        
        const val Outline = 0xFF79747E
        const val OutlineVariant = 0xFFCAC4D0
        const val Scrim = 0xFF000000
        
        const val SurfaceBright = 0xFFFFFBFE
        const val SurfaceDim = 0xFFDED8E1
        const val SurfaceContainer = 0xFFF3EDF7
        const val SurfaceContainerHigh = 0xFFECE6F0
        const val SurfaceContainerHighest = 0xFFE6E0E9
        const val SurfaceContainerLow = 0xFFF7F2FA
        const val SurfaceContainerLowest = 0xFFFFFFFF
    }
    
    object DarkColors {
        const val Primary = 0xFFD0BCFF
        const val OnPrimary = 0xFF381E72
        const val PrimaryContainer = 0xFF4F378B
        const val OnPrimaryContainer = 0xFFEADDFF
        const val InversePrimary = 0xFF6750A4
        
        const val Secondary = 0xFFCCC2DC
        const val OnSecondary = 0xFF332D41
        const val SecondaryContainer = 0xFF4A4458
        const val OnSecondaryContainer = 0xFFE8DEF8
        
        const val Tertiary = 0xFFEFB8C8
        const val OnTertiary = 0xFF492532
        const val TertiaryContainer = 0xFF633B48
        const val OnTertiaryContainer = 0xFFFFD8E4
        
        const val Error = 0xFFF2B8B5
        const val OnError = 0xFF601410
        const val ErrorContainer = 0xFF8C1D18
        const val OnErrorContainer = 0xFFF9DEDC
        
        const val Background = 0xFF1C1B1F
        const val OnBackground = 0xFFE6E1E5
        
        const val Surface = 0xFF1C1B1F
        const val OnSurface = 0xFFE6E1E5
        const val SurfaceVariant = 0xFF49454F
        const val OnSurfaceVariant = 0xFFCAC4D0
        const val SurfaceTint = 0xFFD0BCFF
        const val InverseSurface = 0xFFE6E1E5
        const val InverseOnSurface = 0xFF313033
        
        const val Outline = 0xFF938F99
        const val OutlineVariant = 0xFF49454F
        const val Scrim = 0xFF000000
        
        const val SurfaceBright = 0xFF3B383E
        const val SurfaceDim = 0xFF141218
        const val SurfaceContainer = 0xFF211F26
        const val SurfaceContainerHigh = 0xFF2B2930
        const val SurfaceContainerHighest = 0xFF36343B
        const val SurfaceContainerLow = 0xFF1D1B20
        const val SurfaceContainerLowest = 0xFF0F0D13
    }
    
    object Spacing {
        const val xs = 4
        const val sm = 8
        const val md = 16
        const val lg = 24
        const val xl = 32
        const val xxl = 48
    }
    
    object Typography {
        // Font sizes in sp
        const val displayLarge = 57
        const val displayMedium = 45
        const val displaySmall = 36
        const val headlineLarge = 32
        const val headlineMedium = 28
        const val headlineSmall = 24
        const val titleLarge = 22
        const val titleMedium = 16
        const val titleSmall = 14
        const val bodyLarge = 16
        const val bodyMedium = 14
        const val bodySmall = 12
        const val labelLarge = 14
        const val labelMedium = 12
        const val labelSmall = 11
    }
    
    object Motion {
        // Animation durations in milliseconds
        const val durationInstant = 0
        const val durationFast = 100
        const val durationMedium = 300
        const val durationSlow = 500
        
        // Easing curves (for platform-specific interpolators)
        const val easingStandard = "cubic-bezier(0.4, 0.0, 0.2, 1)"
        const val easingDecelerate = "cubic-bezier(0.0, 0.0, 0.2, 1)"
        const val easingAccelerate = "cubic-bezier(0.4, 0.0, 1, 1)"
    }
}

File: ios/iosApp/DesignSystem/DesignTokens.swift
import Foundation

// Fallback design tokens for standalone iOS target.
enum DesignTokens {
    struct LightColors {
        let Primary: Int = 0xFF007AFF
        let OnPrimary: Int = 0xFFFFFFFF
        let PrimaryContainer: Int = 0xFFE5F1FF
        let OnPrimaryContainer: Int = 0xFF0A2A4A

        let Secondary: Int = 0xFF5E5CE6
        let OnSecondary: Int = 0xFFFFFFFF
        let SecondaryContainer: Int = 0xFFE8E7FF
        let OnSecondaryContainer: Int = 0xFF20204A

        let Tertiary: Int = 0xFF34C759
        let OnTertiary: Int = 0xFFFFFFFF
        let TertiaryContainer: Int = 0xFFE8FAED
        let OnTertiaryContainer: Int = 0xFF0D3A1D

        let Background: Int = 0xFFFFFFFF
        let OnBackground: Int = 0xFF111111
        let Surface: Int = 0xFFF6F6F7
        let OnSurface: Int = 0xFF111111
        let SurfaceVariant: Int = 0xFFECEDEF
        let OnSurfaceVariant: Int = 0xFF4A4D55

        let Error: Int = 0xFFFF3B30
        let OnError: Int = 0xFFFFFFFF
        let ErrorContainer: Int = 0xFFFFE7E6
        let OnErrorContainer: Int = 0xFF4D120F

        let Outline: Int = 0xFFB7BBC3
        let OutlineVariant: Int = 0xFFD8DBE1
        let InversePrimary: Int = 0xFF8EC8FF
        let InverseSurface: Int = 0xFF1E1F22
        let InverseOnSurface: Int = 0xFFF4F4F5
        let SurfaceTint: Int = 0xFF007AFF
        let Scrim: Int = 0xFF000000

        let SurfaceBright: Int = 0xFFFFFFFF
        let SurfaceDim: Int = 0xFFECEDEF
        let SurfaceContainer: Int = 0xFFF2F3F5
        let SurfaceContainerHigh: Int = 0xFFE9EBEE
        let SurfaceContainerHighest: Int = 0xFFE1E4E8
        let SurfaceContainerLow: Int = 0xFFF7F8F9
        let SurfaceContainerLowest: Int = 0xFFFFFFFF
    }

    struct DarkColors {
        let Primary: Int = 0xFF4DA3FF
        let OnPrimary: Int = 0xFF001A33
        let PrimaryContainer: Int = 0xFF003B70
        let OnPrimaryContainer: Int = 0xFFD6E9FF

        let Secondary: Int = 0xFF8A88FF
        let OnSecondary: Int = 0xFF18173D
        let SecondaryContainer: Int = 0xFF2D2A60
        let OnSecondaryContainer: Int = 0xFFE1DFFF

        let Tertiary: Int = 0xFF69D98A
        let OnTertiary: Int = 0xFF093116
        let TertiaryContainer: Int = 0xFF14552A
        let OnTertiaryContainer: Int = 0xFFD7F6E1

        let Background: Int = 0xFF0F1012
        let OnBackground: Int = 0xFFECEDEF
        let Surface: Int = 0xFF17191C
        let OnSurface: Int = 0xFFECEDEF
        let SurfaceVariant: Int = 0xFF2A2D33
        let OnSurfaceVariant: Int = 0xFFC4C8D0

        let Error: Int = 0xFFFF6E66
        let OnError: Int = 0xFF3B0A07
        let ErrorContainer: Int = 0xFF5E1310
        let OnErrorContainer: Int = 0xFFFFDAD7

        let Outline: Int = 0xFF8F949F
        let OutlineVariant: Int = 0xFF3D4149
        let InversePrimary: Int = 0xFF007AFF
        let InverseSurface: Int = 0xFFECEDEF
        let InverseOnSurface: Int = 0xFF232528
        let SurfaceTint: Int = 0xFF4DA3FF
        let Scrim: Int = 0xFF000000

        let SurfaceBright: Int = 0xFF2A2D33
        let SurfaceDim: Int = 0xFF111315
        let SurfaceContainer: Int = 0xFF1B1D21
        let SurfaceContainerHigh: Int = 0xFF23262B
        let SurfaceContainerHighest: Int = 0xFF2B2F35
        let SurfaceContainerLow: Int = 0xFF17191C
        let SurfaceContainerLowest: Int = 0xFF0E1012
    }

    struct Spacing {
        let xs: Double = 4
        let sm: Double = 8
        let md: Double = 12
        let lg: Double = 16
        let xl: Double = 20
        let xxl: Double = 24
    }

    struct Typography {
        let displayLarge: Int32 = 57
        let displayMedium: Int32 = 45
        let displaySmall: Int32 = 36
        let headlineLarge: Int32 = 32
        let headlineMedium: Int32 = 28
        let headlineSmall: Int32 = 24
        let titleLarge: Int32 = 22
        let titleMedium: Int32 = 16
        let titleSmall: Int32 = 14
        let bodyLarge: Int32 = 16
        let bodyMedium: Int32 = 14
        let bodySmall: Int32 = 12
        let labelLarge: Int32 = 14
        let labelMedium: Int32 = 12
        let labelSmall: Int32 = 11
    }
}
